src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageLocalUtilityImpl.java

```diff
@@ -220,4 +220,48 @@ public String getFileExtension(List<? extends SequencingObject> sequencingObject
 		}
 		return bytes;
 	}
+
+	@Override
+	public boolean checkConnectivity() {
+		// Since it's the local filesystem we just return true
+		return true;
+	}
+
+	@Override
+	public boolean checkWriteAccess(Path baseDirectory) {
+		if (!Files.exists(baseDirectory)) {
+			logger.error("Cannot continue startup; base directory " + baseDirectory + " does not exist!");
+			System.exit(1);
+		} else {
+			try {
+				// Check if basedirectory path is writeable by creating a temp file and then removing it
+				Path tempFile = Files.createTempFile(baseDirectory, "", "");
+				// Check if directory is writeable
+				boolean directoryWriteable = Files.isWritable(tempFile);
+
+				try {
+					// Cleanup the temp file created in the directory
+					Files.delete(tempFile);
+				} catch (IOException e) {
+					logger.error("An I/O error occurred while attempting to remove temp file ", e);
```

Comment:
This is probably not an error we just want to swallow.  This should prevent the server from starting.

Reply:
Updated to throw exception in b47ef804ace2d29d02bac15593fc7f7e5129e205